node/metered-channel/src/lib.rs

```diff
@@ -71,114 +98,51 @@ impl Meter {
 	fn note_received(&self) {
 		self.received.fetch_add(1, Ordering::Relaxed);
 	}
+
+	fn note_time_of_flight(&self, tof: Duration) {
+		let _ = self.tof.force_push(tof);
+	}
 }
 
-#[cfg(test)]
-mod tests {
-	use super::*;
-	use futures::{executor::block_on, StreamExt};
+/// Determine if this instance shall be measured
+#[inline(always)]
+fn measure_tof_check(nth: usize) -> bool {
+	const TOF_MEASURE_EVERY_NTH: usize = if cfg!(test) { 2 } else { 16 };
```

Comment:
IMO random sampling would serve us better here and would avoid patterns in the data

Reply:
At the least I'd suggest using a prime number for the constant here

Reply:
Pushing commit

Reply:
I don't see that commit yet.

Reply:
sorry sir, forgot to push